Modules/_collectionsmodule.c

```diff
@@ -118,9 +118,23 @@ static PyTypeObject deque_type;
 #define CHECK_NOT_END(link)
 #endif
 
+/* A simple freelisting scheme is used to minimize calls to the memory
+   allocator.  It accommodates common use cases where new blocks are being
+   added at about the same rate as old blocks are being freed.
+ */
+
+#define MAXFREEBLOCKS 16
+static Py_ssize_t numfreeblocks = 0;
+static block *freeblocks[MAXFREEBLOCKS];
```

Comment:
Can you define the freelist in PyInterpreterState to make it per-interpreter, and so compatible with subinterpreters?

Reply:
Subinterpreters is not approved a likely won't be for Python 3.10, so I don't want to pay that cost now. These macros are used heavily throughout the rest of the module and are performance sensitive.